### ISSUE TITLE
Shift and shift+tab focus items of the body when modal is open

### ISSUE THREAD

**Issue by dsolovieva** [reactstrap/reactstrap#1679]
- components: `modal`
- reactstrap version `#6.4.0`
- import method `es`
- react version `#16.4.0`
- bootstrap version `#4.1.1`

### What is happening?

If you open the modal, focus any browser element (e.g. address bar), and start to press the tab key to move the focus forward body elements get the focus as usual.

### What should be happening?

As described in https://www.w3.org/TR/wai-aria-practices/#dialog_modal elements outside the modal dialog should not be interactive:

> Windows under a modal dialog are inert. That is, users cannot interact with content outside an active dialog window. 

Additionally, here is an example of implementation: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

### Steps to reproduce the issue

1. Go to the http://reactstrap.github.io/components/modals/
2. Open the modal dialog
3. Set focus to the address bar of the browser
4. Press tab until focus goes inside the page

**Comment by TheSharpieOne** [comment]
I could not reproduce the issue given your steps. It looks like you are using an old version of reactstrap while the docs are on the latest version.
The version you are using is 6.3.0, in 6.4.0 the trapping focus was added (see the [release notes](https://github.com/reactstrap/reactstrap/releases/tag/6.4.0))

![modal-focus-trap](https://user-images.githubusercontent.com/664714/67229748-f0012280-f409-11e9-8063-3f6fd8787f20.gif)

**Comment by dsolovieva** [comment]
I use  6.4.0 version and the issue is reproducible on the docs page. It
seems you missed the third step "Set focus to the address bar of the
browser". This is the screencast from the docs page:
[image: reactstrap_modal.gif]
<https://mail.google.com/mail/u/1?ui=2&ik=8b84160024&attid=0.1&permmsgid=msg-a:r-5147660821919087212&view=att&disp=safe&realattid=f_k21lyazw0>

пн, 21 окт. 2019 г. в 21:52, Evan Sharp <
<https://mail.google.com/mail/u/1?ui=2&ik=8b84160024&attid=0.1&permmsgid=msg-a:r-5147660821919087212&view=att&disp=safe&realattid=f_k21lyazw0>
notifications@github.com>:

**Comment by TheSharpieOne** [comment]
It looks like bootstrap is capturing all of the focus events and looking to see if the element focused is within the modal or not. If the element focused is not within the modal, it will focus the modal. This works for when tabbing from the address bar, but is completely broken when shift-tabbing. Bootstrap is planning a new implementation with v5.
For now, I think their approach has some merit. It, combined with the what we currently have should fix this issue as well as continue to work for shift-tabbing.

**Comment by waqasnasir** [comment]
can I work on this?

**Comment by watinha** [comment]
Hey guys, if there is anything not ok in this fix, please let me know.
